envoy/service/discovery/v2/eds.proto

```diff
@@ -0,0 +1,142 @@
+syntax = "proto3";
+
+package envoy.service.discovery.v2;
+
+import "envoy/service/discovery/v2/common.proto";
+import "envoy/api/v2/base.proto";
+import "envoy/api/v2/health_check.proto";
+
+import "google/api/annotations.proto";
+import "google/protobuf/wrappers.proto";
+
+import "validate/validate.proto";
+
+// [#protodoc-title: Endpoints and EDS]
+
+// [#not-implemented-hide:] Not configuration. Workaround c++ protobuf issue with importing
+// services.
+message EdsDummy {
+}
+
+service EndpointDiscoveryService {
+  // The resource_names field in DiscoveryRequest specifies a list of clusters
+  // to subscribe to updates for.
+  rpc StreamEndpoints(stream DiscoveryRequest) returns (stream DiscoveryResponse) {
+  }
+
+  rpc FetchEndpoints(DiscoveryRequest) returns (DiscoveryResponse) {
+    option (google.api.http) = {
+      post: "/v2/discovery:endpoints"
+      body: "*"
+    };
+  }
+}
+
+// An Endpoint that Envoy can route traffic to.
+message LbEndpoint {
```

Comment:
I think LbEndpoint (and below LocalityLbEndpoints) are fairly core concepts in the API, so belong over in `envoy.api.v2.endpoint`. CluserLoadAssignment can probably stay as it's only used as a response message.

Reply:
Yes, I like that. Will apply that.